api/tests/Unit/State/VersionProviderTest.php

```diff
@@ -21,174 +21,128 @@ protected function setUp(): void
         $this->provider = new VersionProvider($this->downloaderFactory);
     }
 
-    public function testProvideCollectionReturnsAllVersionFields(): void
+    public function testProvideCollectionReturnsVersionsWithLatestVersion(): void
     {
-        $downloader1 = $this->createMockDownloader('youtube-dl', '2024.06.01', '2024.07.01');
-        $downloader2 = $this->createMockDownloader('gallery-dl', '1.25.0', '1.26.0');
+        $mockDownloader1 = $this->createMock(DownloaderInterface::class);
+        $mockDownloader1->method('getIdentifier')->willReturn('downloader1');
+        $mockDownloader1->method('getCurrentVersion')->willReturn('1.0.0');
+        $mockDownloader1->method('getLatestVersion')->willReturn('1.1.0');
 
-        $this->downloaderFactory
+        $mockDownloader2 = $this->createMock(DownloaderInterface::class);
+        $mockDownloader2->method('getIdentifier')->willReturn('downloader2');
+        $mockDownloader2->method('getCurrentVersion')->willReturn('2.0.0');
+        $mockDownloader2->method('getLatestVersion')->willReturn('2.5.0');
+
+        $this->downloaderFactory->expects($this->once())
             ->method('getEnabledDownloaders')
-            ->willReturn([$downloader1, $downloader2]);
+            ->willReturn([$mockDownloader1, $mockDownloader2]);
 
-        $operation = $this->createCollectionOperation();
+        // Mock CollectionOperationInterface instead of final GetCollection class
+        $operation = $this->createMock(CollectionOperationInterface::class);
```

Comment:
`VersionProvider->provide()` expects `Operation` (or one of it's descendants).
While `GetCollection` DOES inherit `Operation`, `Operation` does NOT implement the `CollectionOperationInterface`, you should mock the `GetCollection` class here as it currently fails unit tests.